apps/website/src/content/docs/themeprovider.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: ThemeProvider
+description: # TODO
+thumbnail: # TODO
+---
+
+ThemeProvider must be used as an ancestor of all iTwinUI components. It provides theming and other necessary information to all components within the tree.
+
+```jsx
+<ThemeProvider>
+  <>{/* Your components go here. */}</>
+</ThemeProvider>
+```
+
+It is important to note that this component is not just a [context provider](https://react.dev/reference/react/createContext#provider), but also renders an actual DOM element. This is necessary for the theme to be [scoped](#scoping-and-nesting) within the tree.
+
+## Theming
+
+The `theme` prop can be set to either `"light"` or `"dark"`. This will automatically and performantly update the theme for the entire tree, by changing the values of [CSS variables](variables).
+
+```jsx
+<ThemeProvider theme='dark'>…</ThemeProvider>
+```
+
+This theme affects all components that use iTwinUI's CSS variables, including custom components and components from other packages.
+
+If you want to set the theme based on the user's operating system, you can use the `"os"` value. This relies on the [`prefers-color-scheme` query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) to be executed within client-side JavaScript, so it may cause a [flash of incorrect theme](https://css-tricks.com/flash-of-inaccurate-color-theme-fart/) when using server-side rendering.
+
+```jsx
+<ThemeProvider theme='os'>…</ThemeProvider>
+```
+
+If a `theme` value is not specified, it will default to [`"inherit"`](#inheritance), which will inherit the theme from its ancestors and fall back to `"light"`.
+
+### High contrast mode
+
+To enable higher contrast versions of light and dark themes, `themeOptions.highContrast` can be used. By default, the [`prefers-contrast` query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast) will be used to automatically enable high contrast themes if the user has a specified a preference for it in their operating system.
+
+```jsx
+<ThemeProvider
+  theme='dark'
+  themeOptions={{
+    highContrast: true,
+  }}
+/>
+```
+
+### Background
+
+By default, the topmost `ThemeProvider` in the tree will set the `background-color` to `var(--iui-color-background-backdrop)`, which is the recommended value for page background.
+
+If you want to explicitly control whether or not the backgrond is applied, you can override this behavior using `themeOptions.applyBackground`:
+
+```jsx
+<ThemeProvider
+  themeOptions={{
+    applyBackground: false, // or true
```

Comment:
the default is not true though. `true` or `false` will override the default.

Reply:
Ohh, I guess reading the first two sentence of this paragraph gave me this impression. Maybe we can add an "only" to make it more clear?

<img width="685" alt="Screenshot 2023-12-20 at 10 25 03 AM" src="https://github.com/iTwin/iTwinUI/assets/45748283/fc213646-d82a-404b-b9bc-8be8862bec18">

```diff
- By default, the topmost `ThemeProvider` in the tree will set the `background-color` to `var(--iui-color-background-backdrop)`
+ By default, only the topmost `ThemeProvider` in the tree will set the `background-color` to `var(--iui-color-background-backdrop)` (similar to setting `applyBackground` to `true`)
```

Reply:
i'm not sure about the parenthetical at the end, but i've added the word "only"

Reply:
> i'm not sure about the parenthetical at the end

I feel that it is currently not very clear that `applyBackground={true}` sets the background color to background-backdrop. It currently could mean that only the topmost provider sets the background color to background-backdrop. So maybe this can be made clearer some other way.

Reply:
a little verbose but see [`f087260`](https://github.com/iTwin/iTwinUI/pull/1745/commits/f0872605d5241c034fbbd2c773be217b798187ed)

Reply:
That's good, but yes, it does feel a bit verbose. How about something like this?

---

The `themeOptions.applyBackground` prop can be used to control whether or not the `background-color` is set to `var(--iui-color-background-backdrop)` (the recommended value for the page background).

By default, `applyBackground` is automatically set to `true` in only the topmost `ThemeProvider` in the tree.

Passing the `themeOptions.applyBackground` takes precedence over this default behavior.

<details>
<summary>Plain text</summary>

```
The `themeOptions.applyBackground` prop can be used to control whether or not the `background-color` is set to `var(--iui-color-background-backdrop)` (the recommended value for the page background).

By default, `applyBackground` is automatically set to `true` in only the topmost `ThemeProvider` in the tree.

Passing the `themeOptions.applyBackground` takes precedence over this default behavior.
```

</details>

Reply:
I'm not a fan of this and it's not much less verbose.

I like describing the default behavior first and then describing the prop used for overriding it (rather than describing the prop first and describing the default value of the prop). Ideally, most users should not even need to know about the prop, they only care about the behavior.

Reply:
Ohh, I see, then [`f087260` (#1745)](https://github.com/iTwin/iTwinUI/pull/1745/commits/f0872605d5241c034fbbd2c773be217b798187ed) or even the version prior to that would be good.

Also, I noticed a typo in word "background" in both versions.

Reply:
fixed